### PR TITLE
Improve task definition

### PR DESCRIPTION
**Pull request by tshemsedinov**

### CODE REVIEW

File: doc/Task.md [hunk 1]
```diff
@@ -4,36 +4,38 @@
...
 - Task fields:
-  - `caption`
-  - `execute` allowed values: `single`, `server`, `instance`
-  - `time` (optional) defines certain time in UTC
-  - `interval` (optional) defines interval in milliseconds
-  - `run` is an async function to execute
+  - `caption` - task display name
+  - `execute` - execution type, allowed values: `single`, `server`, `instance`
+  - `time` (optional) - defines certain time in UTC
+  - `interval` (optional) - defines interval in milliseconds
```
Comment (lundibundi) [inline doc/Task.md:10]: It is not clear what `time` and `interval` mean here.
I assume `time` means execute at a certain time [in the future], but this doesn't include allowed formats nor specify what will happen if we schedule a `execute: 'single'` task in the past.

I assume `interval` means 'execute every interval millis' but we also have non-optional `execute` parameter that means 'execute once per', what will it be if the `interval` is passed (which implies multiple executions).

File: doc/Task.md [hunk 2]
```diff
@@ -4,36 +4,38 @@
...
-  - `lastStart` last execution start time or `null` (never)
-  - `lastEnd` last execution end time or `null` (never)
-  - `executing` current status: `true`, `false`
-  - `active` is active in current process: `true`, `false`
-  - `count` executing count `number`
+  - `name` - file name without extension used to access tasks
+  - `success` - last execution status: `undefined`, `true`, `false`
+  - `error` - last error or `null` (if success)
+  - `lastStart` - last execution start time or `null` (never)
+  - `lastEnd` - last execution end time or `null` (never)
```
Comment (lundibundi) [inline doc/Task.md:18]: Here `null` is used to define 'no-value' but the `success` uses `undefined` for that purpose.

File: doc/Task.md [hunk 3]
```diff
@@ -4,36 +4,38 @@
...
+  - `executing` - current status: `true`, `false`
+  - `active` - active in current process: `true`, `false`
```
Comment (lundibundi) [inline doc/Task.md:20]: IMO It is not clear what is the difference between `executing` and `active` and what does `active` mean here.

**Comment by tshemsedinov** [comment]
Closed in favour of https://github.com/metarhia/Contracts/pull/47
Implemented in https://github.com/metarhia/impress/releases/tag/v2.5.0

**Pull request closed without merging**